File: BehaviorTreesLib/Sequence.cpp
#include "Sequence.h"

namespace fluentBehaviorTree
{
	Sequence::Sequence(std::string name)
	{
		this->setName(name);
	}

	// Propagate tick to children. Return SUCCESS if no child fails
	EStatus Sequence::tickNode()
	{
		if (this->getResult() == EStatus::RUNNING)
		{
			for each(auto n in mChildren)
			{
				if (!n->isOpen() && n->getResult() == EStatus::FAILURE)
				{
					this->setResult(EStatus::FAILURE);
					return this->getResult();
				}
				else if (n->getResult() == EStatus::RUNNING)
				{
					this->setResult(n->tick());
					if (this->getResult() != EStatus::SUCCESS) return this->getResult();
				}
			}
			this->setResult(EStatus::SUCCESS);
		}
		return this->getResult();
	}
}

File: C#/BehaviorTree/Branch/Composite/Sequence.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FluentBehaviorTree
{
    public class Sequence : Composite
    {
        public Sequence(string name)
        {
            this.Name = name;
            children = new List<Node>();
        }

        /// <summary>
        /// Propagate tick to children. Return SUCCESS if no child fails
        /// </summary>
        /// <returns></returns>
        protected override Status tick()
        {
            if (this.Result == Status.RUNNING)
            {
                foreach (var n in children)
                {
                    if (!n.IsOpen && n.Result == Status.FAILURE)
                    {
                        this.Result = Status.FAILURE;
                        return this.Result;
                    }
                    else if (n.Result == Status.RUNNING)
                    {
                        this.Result = n.Tick();
                        if (this.Result != Status.SUCCESS) return this.Result;
                    }
                }
                this.Result = Status.SUCCESS;
            }
            return this.Result;
        }
    }
}
